cmd/kube-proxy/app/server.go

```diff
@@ -237,6 +237,9 @@ func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, err
 		glog.V(0).Info("Tearing down pure-iptables proxy rules.")
 		iptables.CleanupLeftovers(iptInterface)
 	}
+	if err != nil && proxier == nil {
```

Comment:
I'm confused.  This err is outside the scope of *.NewProxier() calls, and it was previously checked to be not nil (lines 187-188), so it is guaranteed non-nil here.  If proxier is nil, that would be pretty bad.  Did you actually see this fire?